mirror/src/public/main.rs

```diff
@@ -282,8 +294,14 @@ fn rocket() -> _ {
     let mut server = server_builder
         .build()
         .expect("Failed to build mirror GRPC server");
+
     server.start();
```

Comment:
you could equally use `tokio::task::spawn(async move { server.start() })` then return the `JoinHandle` to hold onto, but it's pretty moot as the `server` seems to already pretty much be a handle.

Reply:
Yeah it seems like there's a number of ways we _could_ do this, and I don't think any of them have any real implication one way or another.. @eranrund do you have any preference whether we use a thread or Arc Mutex?

Reply:
Honestly, I prefer using the `tokio::task::spawn` approach myself, it feels cleaner than using Arc Mutex